hubspot3/engagements.py

```diff
@@ -64,21 +64,39 @@ def update(self, key, data=None, **options):
             "engagements/{}".format(key), data=data, method="PUT", **options
         )
 
-    def get_all(self, **options):
+    def get_all(self, limit=-1, **options):
+        engagement_generator = self.get_all_as_generator(limit=limit,
+                                                         **options)
+        return list(engagement_generator)
+
+    def get_all_as_generator(self, limit=-1, **options):
```

Comment:
I think you could do `limit=None` here, and then check if it exists by line 76.

Reply:
I agree that limit=None is better, but this form was already defined there and is used in all the objects I think it is better to keep it consistent through the code, what do you think?

Reply:
OK, I keep forgetting that this code is a little _weird_.